### PR TITLE
Remove the wc combobox component, it is no longer used

### PR DESCRIPTION
**Pull request by samueljseay**
### Changes proposed in this Pull Request:

This is the final clean up needed to close out https://github.com/woocommerce/woocommerce/issues/46357

### How to test the changes in this Pull Request:

1. CI passing should be enough
2. You could also double check there are no references to this component existing anywhere in the codebase as I did

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Remove @wooocommerce `Combobox` component that is no longer used.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Review by mikejolley — DISMISSED** [review]
There are some types exported by that file still being consumed in places. The CI fails are valid. See:

https://github.com/woocommerce/woocommerce/actions/runs/10612882905/job/29415561141?pr=50975

Thanks

**Comment by samueljseay** [comment]
Whoops I missed that failure. Apologies.

Have moved these files to using the types for the Select that they render now anyway. Will ping for re-review when CI is 🍏

File: plugins/woocommerce-blocks/assets/js/base/components/country-input/CountryInputProps.ts
Comment (opr) [inline plugins/woocommerce-blocks/assets/js/base/components/country-input/CountryInputProps.ts:1]: On line 15 of this file the comment references combobox still, could we update it?

**Review by opr — CHANGES_REQUESTED** [review]
Thanks Sam! Just one tiny nit relating to a comment, and in https://github.com/woocommerce/woocommerce/blob/9f7a6779f6d4134d9bc74b9655dcac6dbe108360/plugins/woocommerce-blocks/assets/js/base/components/country-input/style.scss#L3 I just want to confirm if this import is still needed? 

Side note: I tested the main flows in this PR and all are working fine, so once the above is addressed/confirmed then I will be happy to approve.

**Comment by samueljseay** [comment]
> I just want to confirm if this import is still needed?

@opr Oh right 🤔 I think when searching I thought it was still needed for the filter blocks that use it, but I can see now that import is for the country input. We can remove that :) thanks for pointing that out.

File: plugins/woocommerce-blocks/assets/js/base/components/country-input/CountryInputProps.ts
Comment (samueljseay) [reply plugins/woocommerce-blocks/assets/js/base/components/country-input/CountryInputProps.ts:1]: Thanks good spotting. Fixed.

**Review by opr — APPROVED** [review]
Thanks, Sam this looks great now. Tested in Edge, Firefox, Safari 👌🏼

**Pull request merged**